pkg/build/builder/common.go

```diff
@@ -82,3 +92,124 @@ func updateBuildRevision(c client.BuildInterface, build *api.Build, sourceInfo *
 		glog.Warningf("An error occurred saving build revision: %v", err)
 	}
 }
+
+// randomBuildTag generates a random tag used for building images in such a way
+// that the built image can be referred to unambiguously even in the face of
+// concurrent builds with the same name in the same namespace.
```

Comment:
i'm curious now why we wouldn't use the same naming logic for this tag as we use for the container name?  if it's important that the container name be deterministic, isn't it important this tag name be deterministic too?

And the same reasoning applies about conflicts:  as long as the build-name(including namespace) is part of the tag, there will be no conflicts between concurrent builds.